core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1499,4 +1499,18 @@ class ParserTest extends munit.ScalaCheckSuite {
     }
   }
 
+  property("p.filter(_ => true) == p") {
+    forAll(ParserGen.gen, Arbitrary.arbitrary[String]) { (genP, str) =>
+      val res0 = genP.fa.filter(_ => true).parse(str)
+      val res1 = genP.fa.parse(str)
+      assertEquals(res0, res1)
+    }
+  }
+
+  property("p.filter(_ => false) fails") {
+    forAll(ParserGen.gen, Arbitrary.arbitrary[String]) { (genP, str) =>
+      val res = genP.fa.filter(_ => false).parse(str)
+      assert(res.isLeft)
+    }
+  }
```

Comment:
Can we write a law on branch directly (see the laws I have on how ~ composes). I don't think what we have now is exactly right but since filter is weaker is might not detect it.